CppCoreGuidelines.md

```diff
@@ -7976,6 +7989,24 @@ In this example, `Shape` does not inherit from `Geometric_attributes`. Only its
         view_of_trisyms.emplace_back(trisym);
       }
     }
+    
+##### Example
+    
+    class A { ... };
+    
+    class B { ... };
+    
+    void some_function(B* p) {
+      auto base = dynamic_cast<A*>(p);
+      if (base == nullptr) {
```

Comment:
just a few lines above, in the old example, it uses the C++98 syntax that was made for this purpose
`if(auto base = dynamic_cast<A*>(p)) { /* use base */ }`

what does this add to C.148, anyway? It already has an example.